src/containers/share-modal/components/ShareDrawer.tsx

```diff
@@ -17,24 +29,64 @@ export const ShareDrawer = () => {
   const { toast } = useContext(ToastContext)
   const dispatch = useDispatch()
   const track = useSelector(getTrack)
+  const artist = useSelector((state: CommonState) =>
+    getUser(state, { id: track?.owner_id })
+  )
+  const account = useSelector(getAccountUser)
   const source = useSelector(getSource)
 
+  const getActions = () => {
```

Comment:
Does this need to be wrapped in a `useCallback` to not cause rerenders? Otherwise, love this

Reply:
yeah it definitely could, the dep list is gonna be pretty huge which made me think how much will useCallback really optimize. Was talking with Sebastian a bit about this too, like the most optimized option would probably be doing more something like
```
<ActionDrawer>
  <ActionItem .... onClick={...} />
  {isOwner ? <ActionItem>Share To Tiktok</ActionItem> : null}
  ...
</ActionDrawer>
```
that way we are not having to recompute the array of actions a ton of times, and take advantage of general react pref stuff cause the props are closer to each subcomp.

Just thought i'd mention as an idea for the future! in meantime will do the useCallback

Reply:
That's a really good point re: deps. Love the idea of `<ActionDrawerItem>` - definitely a fan of child components over configs